src/tools/ddsperf/ddsperf.c

```diff
@@ -1381,12 +1384,18 @@ struct dds_stats {
   const struct dds_stat_keyvalue *discarded_bytes;
 };
 
-static bool print_stats (dds_time_t tref, dds_time_t tnow, dds_time_t tprev, struct record_cputime_state *cputime_state, struct record_netload_state *netload_state, struct dds_stats *stats)
+static bool print_stats (dds_time_t tref, dds_time_t tnow, dds_time_t tprev, struct record_cputime_state *cputime_state, struct record_netload_state *netload_state, struct dds_stats *stats, char *filename)
```

Comment:
I don't think it makes much sense to pass in the filename here when the `file` is a global variable. It would be more in line with the rest of the code to open the file in `main`, then use it here. (Or, alternatively, to make `file` local to `main` and pass it in here.)

Reply:
Use the filename in optarg to open the file and use the (global) file descriptor.